base/src/main/java/io/spine/code/proto/FieldTypesProto.java

```diff
@@ -0,0 +1,130 @@
+     * return {@code false} for them.
+     *
+     * @param field
+     *         the descriptor of the field to check
+     * @return {@code true} if field is repeated, {@code false} otherwise
+     */
+    public static boolean isRepeated(FieldDescriptorProto field) {
+        checkNotNull(field);
+        boolean result = field.getLabel() == LABEL_REPEATED && !isMap(field);
+        return result;
+    }
+
+    /**
+     * Checks the Protobuf field and determines it is map field or not.
+     *
+     * <p>If a field is a map it is repeated message with the specific
+     * {@linkplain #getEntryNameFor(FieldDescriptorProto) type}.
+     *
+     * @param field
+     *         the descriptor of the field to check
+     * @return {@code true} if field is map, {@code false} otherwise
+     */
+    public static boolean isMap(FieldDescriptorProto field) {
+        checkNotNull(field);
+        if (field.getLabel() != LABEL_REPEATED) {
+            return false;
+        }
+        if (field.getType() != TYPE_MESSAGE) {
+            return false;
+        }
+        boolean result = field.getTypeName()
+                              .endsWith('.' + getEntryNameFor(field));
+        return result;
+    }
+
+    /**
+     * Constructs the entry name for the map field.
+     *
+     * <p>For example, proto field with name 'word_dictionary' has 'wordDictionary' json name.
+     * Every map field has corresponding entry type.
+     * For 'word_dictionary' it would be 'WordDictionaryEntry'
+     *
+     * @param mapField
+     *         the field to construct entry name
+     * @return the name of the map field
+     */
+    public static String getEntryNameFor(FieldDescriptorProto mapField) {
+        FieldName fieldName = FieldName.of(mapField);
+        return fieldName.toCamelCase() + ENTRY_SUFFIX;
+    }
+
+    /**
+     * Checks the Protobuf field and determines it is message type or not.
+     *
+     * @param fieldDescriptor
+     *         the descriptor of the field to check
+     * @return {@code true} if it is message, {@code false} otherwise
+     */
+    public static boolean isMessage(FieldDescriptorProto fieldDescriptor) {
+        checkNotNull(fieldDescriptor);
+        return fieldDescriptor.getType() == TYPE_MESSAGE;
+    }
+
+    /**
+     * Removes the leading dot from the Protobuf type name passed as {@code String}.
+     *
+     * <p>If there is no leading dots, returns the unmodified parameter.
+     *
+     * @param fieldDescriptor
+     *         the field descriptor whose type name to modify
+     * @return the type name without leading dot
+     */
+    public static String trimTypeName(FieldDescriptorProto fieldDescriptor) {
+        String typeName = fieldDescriptor.getTypeName();
+        checkNotNull(typeName);
+
+        if (typeName.isEmpty()) {
+            return typeName;
+        }
+        // it has a redundant dot in the beginning
```

Comment:
Please remove the comment. If the code is not readable enough, extract some variables/functions in order to make it sound better.